x/programs/cmd/simulator/src/lib.rs

```diff
@@ -99,6 +142,8 @@ pub enum Param {
     String(String),
     Id(Id),
     Key(Key),
+    #[allow(private_interfaces)]
+    TestContext(SimulatorTestContext),
```

Comment:
Why do we have to do this?

Reply:
Because we want the user space to only have access to `TestContext`, `SimulatorTestContext` is private and only meant for the simulator, because struct cannot be externally tagged. See https://github.com/ava-labs/hypersdk/pull/1076#discussion_r1651672553